.pre-commit-config.yaml

```diff
@@ -577,6 +577,14 @@ repos:
         files: \.*example_dags.*\.py$
         exclude: ^airflow/_vendor/
         pass_filenames: true
+      - id: check-xfail-usage-in-test-code
+        language: pygrep
+        name: Don't use xfail in tests
+        description: xfail is not recommended as a test meant to fail is generally not meaningful
+        entry: "@pytest.mark.xfail"
+        pass_filenames: true
+        files: ^tests/.*\.py$
+        exclude: ^airflow/|^docs/|^tests/always/test_project_structure.py|^tests/jobs/test_backfill_job.py
```

Comment:
Why do we need `^airflow` and `^docs` here, if we are only matching `^tests/.*\.py$`?

Reply:
good call out, that was just copy pasta from before, will remove it now :)

Reply:
✅